packages/cli/src/ui/hooks/useKeypress.ts

```diff
@@ -64,8 +70,141 @@ export function useKeypress(
+        return {
+          name: 'c',
+          ctrl: true,
+          meta: alt,
+          shift,
+          paste: false,
+          sequence,
+          kittyProtocol: true,
+        };
+      }
+
+      // Handle other keys as needed
+      return null;
+    };
 
     const handleKeypress = (_: unknown, key: Key) => {
+      // Handle VS Code's backslash+return pattern (Shift+Enter)
+      if (key.name === 'return' && previousSequence === '\\') {
+        // Convert to Shift+Enter
+        onKeypressRef.current({
+          ...key,
+          shift: true,
+          sequence: '\\\r', // VS Code's Shift+Enter representation
+        });
+        previousSequence = '';
+        return;
+      }
+      
+      // Don't pass through backslash if it might be part of VS Code's Shift+Enter
+      // We'll pass it through later if no return follows
+      if (key.sequence === '\\' && !key.name) {
+        previousSequence = key.sequence;
+        // Don't pass through yet - wait to see if return follows
+        return;
+      }
+      
+      // If we had a pending backslash and this isn't return, pass through the backslash first
+      if (previousSequence === '\\' && key.name !== 'return') {
+        onKeypressRef.current({
+          name: '',
+          ctrl: false,
+          meta: false,
+          shift: false,
+          paste: false,
+          sequence: '\\',
+        });
+      }
+
+      // Store current sequence for next check (before other processing)
+      previousSequence = key.sequence;
+
+      // Always pass through Ctrl+C immediately, regardless of protocol state
+      // Check both standard format and Kitty protocol sequence
+      if (
+        (key.ctrl && key.name === 'c') ||
+        key.sequence === `\x1b${KITTY_CTRL_C}`
+      ) {
+        kittySequenceBuffer = '';
+        // If it's the Kitty sequence, create a proper key object
+        if (key.sequence === `\x1b${KITTY_CTRL_C}`) {
+          onKeypressRef.current({
+            name: 'c',
+            ctrl: true,
+            meta: false,
+            shift: false,
+            paste: false,
+            sequence: key.sequence,
+            kittyProtocol: true,
+          });
+        } else {
+          onKeypressRef.current(key);
+        }
+        return;
+      }
+
+      // If Kitty protocol is enabled, handle CSI sequences
+      if (kittyProtocolEnabled) {
+        // If we have a buffer or this starts a CSI sequence
+        if (kittySequenceBuffer || key.sequence.startsWith('\x1b[')) {
+          kittySequenceBuffer += key.sequence;
+
+          // Try to parse the buffer as a Kitty sequence
+          const kittyKey = parseKittySequence(kittySequenceBuffer);
+          if (kittyKey) {
+            kittySequenceBuffer = '';
+            onKeypressRef.current(kittyKey);
+            return;
+          }
+
+          // If buffer doesn't match expected pattern and is getting long, flush it
+          if (kittySequenceBuffer.length > 10) {
```

Comment:
comment on why the magic number 10. Is there some alternate magic number we can use aligned with the longest kitty sequence or something?

Reply:
Done.